src/shapes/circle.class.ts

```diff
@@ -1,232 +1,223 @@
- * @static
- * @memberOf Circle
- * @see: http://www.w3.org/TR/SVG/shapes.html#CircleElement
- */
-Circle.ATTRIBUTE_NAMES = ['cx', 'cy', 'r', ...fabric.SHARED_ATTRIBUTES];
-
-/**
- * Returns {@link Circle} instance from an SVG element
- * @static
- * @memberOf Circle
- * @param {SVGElement} element Element to parse
- * @param {Function} [callback] Options callback invoked after parsing is finished
- * @param {Object} [options] Partial Circle object to default missing properties on the element.
- * @throws {Error} If value of `r` attribute is missing or invalid
- */
-Circle.fromElement = function (element, callback) {
-  const {
-    left = 0,
-    top = 0,
-    radius,
-    ...otherParsedAttributes
-  } = fabric.parseAttributes(element, Circle.ATTRIBUTE_NAMES);
-
-  if (!radius || radius < 0) {
-    throw new Error(
-      'value of `r` attribute is required and can not be negative'
+  /* _TO_SVG_END_ */
+
+  /* _FROM_SVG_START_ */
+  /**
+   * List of attribute names to account for when parsing SVG element (used by {@link Circle.fromElement})
+   * @static
+   * @memberOf Circle
+   * @see: http://www.w3.org/TR/SVG/shapes.html#CircleElement
+   */
+  static ATTRIBUTE_NAMES = ['cx', 'cy', 'r', ...SHARED_ATTRIBUTES];
+
+  /**
+   * Returns {@link Circle} instance from an SVG element
+   * @static
+   * @memberOf Circle
+   * @param {SVGElement} element Element to parse
+   * @param {Function} [callback] Options callback invoked after parsing is finished
+   * @param {Object} [options] Partial Circle object to default missing properties on the element.
+   * @throws {Error} If value of `r` attribute is missing or invalid
+   */
+  static fromElement(element: SVGElement, callback: (circle: Circle) => any) {
+    const {
+      left = 0,
+      top = 0,
+      radius,
+      ...otherParsedAttributes
+    } = parseAttributes(element, Circle.ATTRIBUTE_NAMES);
+
+    if (!radius || radius < 0) {
+      throw new Error(
+        'value of `r` attribute is required and can not be negative'
+      );
+    }
+
+    // this probably requires to be fixed for default origins not being top/left.
+    callback(
+      new Circle({
+        ...otherParsedAttributes,
+        radius,
+        left: left - radius,
+        top: top - radius,
+      })
     );
   }
 
-  // this probably requires to be fixed for default origins not being top/left.
-  callback(
-    new fabric.Circle({
-      ...otherParsedAttributes,
-      radius,
-      left: left - radius,
-      top: top - radius,
-    })
-  );
+  /* _FROM_SVG_END_ */
+
+  /**
+   * Returns {@link Circle} instance from an object representation
+   * @static
+   * @memberOf Circle
+   * @param {Object} object Object to create an instance from
+   * @returns {Promise<Circle>}
+   */
+  static fromObject(object: object): Promise<Circle> {
+    return FabricObject._fromObject(Circle, object);
+  }
+}
+
+export const circleDefaultValues: Partial<TClassProperties<Circle>> = {
+  type: 'circle',
+  radius: 0,
```

Comment:
should we bump radius to 100 as default? creating circles that are unusable by default seems just wrong

Reply:
agreed